terraform/workspaces/infra/s3/s3.tf

```
@@ -106,14 +122,40 @@ data "aws_iam_policy_document" "cdn_bucket_policy" {
   # NOTE: Insecure (non-SSL) requests are allowed for this bucket otherwise requests from Minio fail
 }
 
+data "aws_caller_identity" "current" {}
+
+data "aws_elb_service_account" "main" {}
+
+data "aws_iam_policy_document" "logs_bucket_policy" {
+  count = var.disable_logs ? 0 : 1
+
+  statement {
+    sid     = "AllowPutObjects"
+    actions = ["s3:PutObject"]
+    effect  = "Allow"
+    resources = [
+      "${aws_s3_bucket.logs[0].arn}",
+      "${aws_s3_bucket.logs[0].arn}/AWSLogs/${data.aws_caller_identity.current.account_id}/*",
+    ]
+
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_elb_service_account.main.arn]
+    }
+  }
+}
+
 resource "aws_s3_bucket_policy" "app_bucket" {
   bucket = aws_s3_bucket.app.id
   policy = data.aws_iam_policy_document.app_bucket_policy.json
 }
 
-resource "aws_s3_bucket_policy" "cdn_bucket" {
-  bucket = aws_s3_bucket.cdn.id
-  policy = data.aws_iam_policy_document.cdn_bucket_policy.json
+
+resource "aws_s3_bucket_policy" "logs_bucket" {
+  count = var.disable_logs ? 0 : 1
+
+  bucket = aws_s3_bucket.logs[0].id
+  policy = data.aws_iam_policy_document.logs_bucket_policy[0].json
```

The `0` will cause errors when `aws_s3_bucket.logs` has no items.

```suggestion
  bucket = aws_s3_bucket.logs[count.index].id
  policy = data.aws_iam_policy_document.logs_bucket_policy[count.index].json
```